packages/dropdown/src/js/index.ts

```diff
@@ -50,7 +48,6 @@ interface UseDropdownProps extends Omit<HTMLPropsFor<'button'>, 'onChange'> {
   size?: ValueOf<typeof vars.sizes>
   style?: React.CSSProperties
   subLabel?: ReactNode
-  uniqueId?: (prefix: string) => string
```

Comment:
The mocking pattern you shared should be sufficient for most teams no? Also those id's really do need to be unique for accessibility. We'd have to rely on teams actually adding their own when implementing. I kinda like the id of allowing a data-test-id to be used instead for testing. What do you think?

Reply:
i wish mocking would work but as of yet teams have said they can't do it and need the DI pattern to override in test. mocking probably would work but i haven't dug into it at all - i just know it's the reason that a couple components have a `uniqueId` prop. 

Yes i agree we need the uniqueId so we should still be handling that. the props is for overriding specifically for snapshot testing

Reply:
@EdwardIrby this is where we are allowing a custom `uniqueId` prop to be provided. this is the DI pattern that I was referencing in standup were the `uniqueId` prop can be override in test by consumers

cc @jaketrent thoughts? i know this was recently added to Carousel as well and I see this as an emerging standard. is this something we should continue to support - or is using DI for unique id the wrong approach and we could stepback and rethink

Reply:
My thoughts:

- jest.mock of @ps/ds-util didn't work. By the time we were mocking, the subdep package name was not addressable.
- jest.mock of Math.random does seem like a impl detail. In a way jest.mock of @ps/ds-util is as well, just not as low-level. It's also something we control.
- uniqueId props seem like a simple solution to this problem. 
- I'm not super surprised that "uniqueId" as a name caused confusion.  Maybe it needs something that shows it to be test-centric (eg, TESTONLY_uniqueId) or private (eg, _uniqueId). 
- My favorite solution would be to not have a uniqueId prop. This would require on-page uniqueness and stability between runs. Stability could come from the auto-incrementer (a la downshift) that was mentioned. I like that. But what about on-page uniqueness.  You'd have to auto-increment within a namespace per Dropdown. Where would that uniqueness come from if not a gen'ed uuid? Hash some other data (eg, labels and menu items)?

Reply:
I'm going to put back uniqueID. It is confusing ands seems like a thing I'd need to add as a consumer of the comonent. I honestly wish this was just a data-testid because that's more explicit. But this is blocking other teams so let's keep as is.

Reply:
@jaketrent is there a reason we're putting [formated i'ds](https://github.com/pluralsight/design-system/blob/4f79ccea73ba982bde80c5dfa2b9d05518ba915d/packages/dropdown/src/js/index.ts#L340) on Items. It's not for a11y so I'm just trying to understand the reasoning behind it.